python_modules/dagster-graphql/dagster_graphql/implementation/fetch_assets.py

```diff
@@ -377,6 +378,60 @@ def get_materialized_partitions_subset(
         )
 
 
+def get_failed_partitions_subset(
```

Comment:
can we combine this function with the function above? because we call them sequentially in the resolver, which will make the cache update twice in a row